src/main/java/spoon/support/reflect/declaration/CtPackageImpl.java

```diff
@@ -242,7 +242,12 @@ public boolean hasPackageInfo() {
 
 	@Override
 	public boolean isEmpty() {
-		return getPackages().isEmpty() && getTypes().isEmpty();
+		return getPackages().isEmpty() && !hasTypes();
```

Comment:
Doesn't `getPackages()` have precisely the same problem (allocates a new `LinkedHashSet` on each invocation)? Perhaps kill two birds with one stone in this PR?

Reply:
Yes, `getPackages()` has the same problem, but as it didn't really show up in my profiles (as it's called far less), I didn't include an extra method for the empty check in this PR. If you want, I can do so though.

Reply:
It's probably not a performance problem because the amount of packages is typically far fewer than the amount of types in any given package.

I think from an API standpoint it makes sense to have both. Since there are methods `getTypes()` and `getPackages()`, I probably would expect a `hasTypes()` to be accompanied by a `hasPackages()`. What do you think?